ui/app/components/clients/date-range.hbs

```diff
@@ -0,0 +1,92 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+<div ...attributes>
+  <Hds::Text::Display @tag="p" class="has-bottom-margin-xs">
+    Date range
+  </Hds::Text::Display>
+
+  <div class="is-flex-align-baseline">
+    {{#if (or @startTime @endTime)}}
+      <p class="is-size-6" data-test-date-range="start">{{this.formattedDate @startTime}}</p>
+      <p class="has-left-margin-xs"> — </p>
+      <p class="is-size-6 has-left-margin-xs" data-test-date-range="end">{{this.formattedDate @endTime}}</p>
+    {{else}}
+      <p class="is-size-6" data-test-date-range="default">Using default date range</p>
+    {{/if}}
+    <Hds::Button
+      class="has-left-margin-xs"
+      @text="Edit"
+      @color="tertiary"
+      @icon="edit"
+      @iconPosition="trailing"
+      data-test-date-range-edit
+      {{on "click" (fn (mut this.showEditModal) true)}}
+    />
+  </div>
+
+  {{#if this.showEditModal}}
+    <Hds::Modal data-test-date-range-edit-modal @onClose={{this.onClose}} as |M|>
+      <M.Header>
+        Edit date range
+      </M.Header>
+      <M.Body>
+        <p class="has-bottom-margin-s">
+          The start date will be used as the client counting start time and all clients in that month will be considered new.
+          {{#if this.version.isEnterprise}}
+            We recommend setting this date as your license or billing start date to get the most accurate new and total
+            client count estimations.
+          {{/if}}
```

Comment:
I think when we brought these designs in front of the client counting pod it was suggested we mention that changing these dates does _not_ set or change the user's Vault config. Something like: `These dates are only for querying data in storage. Editing the date range does not change any license or billing configurations.` 🤷 Words are hard!